c10/core/Allocator.h

```diff
@@ -59,6 +59,13 @@ class DataPtr {
   Device device() const {
     return device_;
   }
+  // Unsafely mutates the device on a DataPtr.  Under normal use,
+  // you should never actually need to call this function.
+  // We need this for the implementation of the hack detailed
+  // in Note [Masquerading as CUDA]
+  void unsafe_set_device(Device device) {
```

Comment:
Should we `friend` it instead? If you offer a function, people are going to use it, no matter what the comment says ;)

Reply:
It's a perfectly legitimate function and can be implemented (less conveniently) in user-space, so I don't think it is worth friend'ing.